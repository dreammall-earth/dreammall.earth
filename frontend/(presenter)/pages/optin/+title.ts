import i18n from '#renderer/plugins/i18n'

// 'DreamMall | Newsletter'
export const title = () => i18n.global.t('optin.title')
