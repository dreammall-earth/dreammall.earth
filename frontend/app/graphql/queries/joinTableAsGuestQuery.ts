import { gql } from 'graphql-tag'

export const joinTableAsGuestQuery = gql`
  query ($tableId: Int!, $userName: String!) {
    joinTableAsGuest(tableId: $tableId, userName: $userName)
  }
`
