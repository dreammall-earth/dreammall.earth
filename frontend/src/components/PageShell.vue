<template>
  <v-app>
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
