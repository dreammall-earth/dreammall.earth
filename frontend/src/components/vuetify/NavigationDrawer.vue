<template>
  <v-navigation-drawer v-model="drawer" :location="computedLocation" temporary width="400">
    <slot />
  </v-navigation-drawer>
</template>

<script lang="ts" setup>
import { computed, ref, watch } from 'vue'
import { useDisplay } from 'vuetify'

export type DrawerLocation = 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start'

const props = withDefaults(
  defineProps<{
    modelValue: boolean
    location?: DrawerLocation
  }>(),
  {
    location: 'right',
  },
)

const emits = defineEmits(['update:modelValue'])
const display = useDisplay()
const drawer = ref(props.modelValue)

const computedLocation = computed(() => {
  return display.mobile.value ? 'bottom' : props.location
})

watch(
  () => props.modelValue,
  (newVal) => {
    drawer.value = newVal
  },
)

watch(
  () => drawer.value,
  (newVal) => {
    if (newVal !== props.modelValue) {
      emits('update:modelValue', newVal)
    }
  },
)
</script>
