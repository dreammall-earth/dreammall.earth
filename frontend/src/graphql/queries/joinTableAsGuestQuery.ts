import { gql } from 'graphql-tag'

export const joinTableAsGuestQuery = gql`
  query ($tableId: String!, $userName: String!) {
    joinTableAsGuest(tableId: $tableId, userName: $userName)
  }
`
