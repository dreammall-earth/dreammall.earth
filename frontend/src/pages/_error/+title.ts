export const title = 'DreamMall | Error'
