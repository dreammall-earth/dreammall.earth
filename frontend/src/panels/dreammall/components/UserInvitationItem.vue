<template>
  <v-list-item class="user-invitation-item my-2 pa-2 pr-3 rounded-pill" elevation="0" rounded>
    <template #prepend>
      <v-avatar size="40">
        <v-img v-if="user.avatar" :src="user.avatar" :alt="user.name" cover></v-img>
        <svg v-else width="40" height="40" viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg">
          <circle cx="20" cy="20" r="20" fill="#E0E0E0" />
          <circle cx="20" cy="15" r="7" fill="#BDBDBD" />
          <path
            d="M6 34.5C6 27.5964 11.5964 22 18.5 22H21.5C28.4036 22 34 27.5964 34 34.5V40H6V34.5Z"
            fill="#BDBDBD"
          />
        </svg>
      </v-avatar>
    </template>

    <v-list-item-title class="text-white font-weight-medium">
      {{ user.name }}
    </v-list-item-title>

    <template #append>
      <v-btn
        elevation="0"
        :color="user.invited ? 'success' : 'secondary'"
        class="invitation-button text-caption py-2 px-3 rounded-pill"
        size="small"
        @click="toggleInvitation"
      >
        {{
          user.invited
            ? $t('dream-mall-panel.setup.invitation-sent')
            : $t('dream-mall-panel.setup.invite')
        }}
      </v-btn>
    </template>
  </v-list-item>
</template>

<script lang="ts" setup>
import { defineProps, defineEmits } from 'vue'

import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const props = defineProps<{
  user: UserInvitation
}>()

const emit = defineEmits<{
  (e: 'update:invited', value: boolean): void
}>()

const toggleInvitation = () => {
  emit('update:invited', !props.user.invited)
}
</script>

<style lang="scss" scoped>
.user-invitation-item {
  --background-user-item: #5d6670; // todo: save globally
  --background-invitation-button: #8b949b; // todo: save globally

  background-color: var(--background-user-item) !important;

  .invitation-button {
    &.bg-secondary {
      background-color: var(--background-invitation-button) !important;
    }
  }
}
</style>
