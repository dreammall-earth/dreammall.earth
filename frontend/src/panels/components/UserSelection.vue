<template>
  <v-text-field
    v-model="userSearch"
    rounded
    clearable
    class="elevation-0 w-100"
    content-class="elevation-0"
    label="Suche"
    variant="solo-filled"
    append-inner-icon="mdi-magnify"
  />

  <div class="user-list-container w-100">
    <div v-if="error" class="text-center text-error">
      {{ $t('dream-mall-panel.setup.search-error') }}
    </div>

    <v-list
      v-if="displayedUsers.length"
      class="bg-transparent w-100 user-list"
      :class="{ loading: isLoading }"
    >
      <UserInvitationItem
        v-for="user in displayedUsers"
        :key="user.id"
        :user="user"
        @update:invited="updateInvitationStatus(user.id, $event)"
      />
    </v-list>

    <div v-if="!displayedUsers.length" class="text-center mt-4 text-grey-darken-2">
      {{ $t('dream-mall-panel.setup.no-result') }}
    </div>
  </div>
</template>

<script lang="ts" setup>
import { ref, computed, watch } from 'vue'

import UserInvitationItem from '#src/panels/components/UserInvitationItem.vue'
import useSearchUsers from '#src/panels/composables/useSearchUsers'
import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const users = defineModel<number[]>({ required: true })

const { searchUsers, searchResults, isLoading, error } = useSearchUsers()
const userSearch = ref('')

watch(userSearch, (newSearch) => {
  searchUsers(newSearch)
})

const displayedUsers = computed<UserInvitation[]>(() => {
  return searchResults.value.map((user) => ({
    id: user.id,
    name: user.name || user.username,
    avatar: null, // not supporting yet
    invited: users.value.includes(user.id), // todo: user already included in myTable?
  }))
})

const updateInvitationStatus = (userId: number, invited: boolean) => {
  let currentUsers: number[] = [...users.value]
  if (invited && !currentUsers.includes(userId)) {
    currentUsers.push(userId)
  } else if (!invited && currentUsers.includes(userId)) {
    currentUsers = currentUsers.filter((id) => id !== userId)
  }
  users.value = currentUsers
}
</script>

<style lang="scss" scoped>
.flat-text-field .v-field--variant-solo-filled {
  box-shadow: none !important;
}

.user-list-container {
  height: 200px;
  overflow-y: auto;

  .user-list {
    height: 100%;

    &.loading {
      opacity: 0.5;
      transition: opacity 0.5s ease;
    }
  }
}
</style>
