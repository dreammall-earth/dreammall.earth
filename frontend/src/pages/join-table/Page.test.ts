import { ApolloError } from '@apollo/client/errors'
import { provideApolloClient } from '@vue/apollo-composable'
import { flushPromises, mount } from '@vue/test-utils'
import { createMockClient } from 'mock-apollo-client'
import { describe, it, expect, vi, beforeEach } from 'vitest'
import { Component, h } from 'vue'
import { VApp } from 'vuetify/components'

import { joinTableAsGuestQuery } from '#queries/joinTableAsGuestQuery'
import { errorHandlerSpy } from '#tests/plugin.globalErrorHandler'

import JoinTablePage from './+Page.vue'
import Route from './+route'
import { title } from './+title'

const joinTableAsGuestQueryMock = vi.fn()

const mockClient = createMockClient()

mockClient.setRequestHandler(
  joinTableAsGuestQuery,
  joinTableAsGuestQueryMock.mockResolvedValue({ data: { joinTable: 'http://some.url' } }),
)

provideApolloClient(mockClient)

describe('JoinTablePage', () => {
  const Wrapper = () => {
    return mount(VApp, {
      slots: {
        default: h(JoinTablePage as Component),
      },
    })
  }

  let wrapper: ReturnType<typeof Wrapper>

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('returns join Table title', () => {
    expect(title()).toBe('Raum beitreten')
  })

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('has tableID as param', () => {
    expect(Route).toBe('/join-table/@id')
  })

  describe('Api', () => {
    beforeEach(async () => {
      vi.clearAllMocks()
      await wrapper.find('input').setValue('PinkyPie')
      await wrapper.find('form').trigger('submit')
    })

    it('calls JoinTable query', () => {
      // eslint-disable-next-line vitest/prefer-called-with
      expect(joinTableAsGuestQueryMock).toHaveBeenCalled()
    })

    describe('Table Link returned', () => {
      beforeEach(async () => {
        joinTableAsGuestQueryMock.mockResolvedValue({
          data: { joinTableAsGuest: 'http://meinlink.de' },
        })
        vi.clearAllMocks()
        await wrapper.find('form').trigger('submit')
      })

      it('Redirects to table Link', () => {
        expect(global.window.location.href).toBe('http://meinlink.de/')
      })
    })

    describe('Error returned', () => {
      beforeEach(async () => {
        joinTableAsGuestQueryMock.mockRejectedValue({ message: 'autsch' })
        await flushPromises()
        vi.clearAllMocks()
        await wrapper.find('form').trigger('submit')
      })

      it('logs Table not found', () => {
        expect(errorHandlerSpy).toHaveBeenCalledWith(
          'table link not found',
          new ApolloError({ errorMessage: 'autsch' }),
        )
      })
    })
  })
})
