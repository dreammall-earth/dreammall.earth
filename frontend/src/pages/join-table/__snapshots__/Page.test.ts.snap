// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JoinTablePage > renders 1`] = `
<div
  class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr"
>
  <div
    class="v-application__wrap"
  >
    
    <div
      class="background"
      data-v-bb2bf2be=""
    >
      <form
        class="panel"
        data-v-bb2bf2be=""
      >
        <div
          class="d-flex flex-column align-center justify-space-around pa-4"
          data-v-bb2bf2be=""
        >
          <h2
            class="section-headline"
            data-v-bb2bf2be=""
          >
            Test Table
          </h2>
          <div
            class="reminder text-center mt-8 pa-5 font-weight-medium"
            data-v-bb2bf2be=""
          >
            <div
              class="v-responsive v-img v-img--booting w-100 logo-tiny mx-auto"
              data-v-bb2bf2be=""
              data-v-f6be2607=""
            >
              <div
                class="v-responsive__sizer"
              />
              
              
              <transition-stub
                appear="true"
                css="true"
                name="fade-transition"
                persisted="false"
              >
                <img
                  class="v-img__img v-img__img--contain"
                  src="/src/assets/dreammall-logo-simple.svg"
                  style="display: none;"
                />
              </transition-stub>
              <transition-stub
                appear="false"
                css="true"
                name="fade-transition"
                persisted="false"
              >
                <!---->
              </transition-stub>
              <!---->
              <!---->
              <!---->
              
              
              <!---->
            </div>
            <p
              class="mt-5"
              data-v-bb2bf2be=""
            >
              $t('joinTablePage.reminder')
            </p>
          </div>
          <div
            class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field elevation-0 w-75 flex-grow-0 mt-12"
            data-v-bb2bf2be=""
          >
            <!---->
            <div
              class="v-input__control"
            >
              
              <div
                class="v-field v-field--appended v-field--center-affix v-field--flat v-field--variant-solo-filled v-theme--light v-field--rounded v-locale--is-ltr"
              >
                <div
                  class="v-field__overlay"
                />
                <div
                  class="v-field__loader"
                >
                  <div
                    aria-hidden="true"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    class="v-progress-linear v-theme--light v-locale--is-ltr"
                    role="progressbar"
                    style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
                  >
                    <!---->
                    <div
                      class="v-progress-linear__background"
                      style="opacity: NaN;"
                    />
                    <div
                      class="v-progress-linear__buffer"
                      style="opacity: NaN; width: 0%;"
                    />
                    <transition-stub
                      appear="false"
                      css="true"
                      name="fade-transition"
                      persisted="false"
                    >
                      <div
                        class="v-progress-linear__indeterminate"
                      >
                        
                        <div
                          class="v-progress-linear__indeterminate long"
                        />
                        <div
                          class="v-progress-linear__indeterminate short"
                        />
                        
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div
                  class="v-field__field"
                  data-no-activator=""
                >
                  <label
                    aria-hidden="true"
                    class="v-label v-field-label v-field-label--floating"
                    for="input-0"
                  >
                    <!---->
                    
                    Gastname
                    
                  </label>
                  <label
                    class="v-label v-field-label"
                    for="input-0"
                  >
                    <!---->
                    
                    Gastname
                    
                  </label>
                  
                  
                  <!---->
                  <input
                    aria-describedby="input-0-messages"
                    autofocus=""
                    class="v-field__input"
                    content-class="elevation-0"
                    id="input-0"
                    maxlength="64"
                    required=""
                    size="1"
                    type="text"
                    value=""
                  />
                  <!---->
                  
                  
                </div>
                <!---->
                <div
                  class="v-field__append-inner"
                >
                  <!---->
                  <i
                    aria-hidden="true"
                    class="mdi-pencil mdi v-icon notranslate v-theme--light v-icon--size-default"
                  />
                </div>
                <div
                  class="v-field__outline"
                >
                  <!---->
                  <!---->
                </div>
              </div>
              
            </div>
            <!---->
            <div
              class="v-input__details"
            >
              <transition-group-stub
                appear="false"
                aria-live="polite"
                class="v-messages"
                css="true"
                id="input-0-messages"
                name="slide-y-transition"
                persisted="false"
                role="alert"
                tag="div"
              >
                <!---->
              </transition-group-stub>
              <!---->
            </div>
          </div>
          <!-- todo: manage values as maxlength globally? -->
          <button
            class="v-btn v-btn--elevated v-theme--light v-btn--density-default elevation-0 v-btn--rounded v-btn--size-default v-btn--variant-elevated custom-green-btn mt-12 mx-auto"
            data-v-bb2bf2be=""
            data-v-ea088446=""
            type="button"
          >
            
            <span
              class="v-btn__overlay"
            />
            <span
              class="v-btn__underlay"
            />
            
            <!---->
            <span
              class="v-btn__content"
              data-no-activator=""
            >
              
              
              Tisch betreten
              
              
            </span>
            <!---->
            <!---->
          </button>
        </div>
      </form>
    </div>
    
  </div>
</div>
`;
